catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/MetacardApplication.java

```diff
@@ -624,8 +623,10 @@ protected UpdateResponse patchMetacards(List<MetacardChanges> metacardChanges)
                 .stream()
                 .map(Result::getMetacard)
                 .collect(Collectors.toList());
-        return catalogFramework.update(new UpdateRequestImpl(changedIds.toArray(new String[0]),
-                changedMetacards));
+        return catalogFramework.update(new UpdateRequestImpl(changedMetacards.stream()
+                .map(Metacard::getId)
+                .collect(Collectors.toList())
+                .toArray(new String[0]), changedMetacards));
```

Comment:
According to docs https://docs.oracle.com/javase/8/docs/api/java/util/stream/package-summary.html

Streams that are sourced from things that define order will make the stream intrinsically ordered. So this means using `changedMetacards` for both arguments will result in the order of id's match.